.github/workflows/psalm.yml

```diff
@@ -37,24 +37,51 @@ concurrency:
 jobs:
   static-analysis:
     runs-on: ubuntu-latest
+    strategy:
+      # do not stop on another job's failure
+      fail-fast: false
+      matrix:
+        ocp-version: ['30', '31', 'dev-master']
+        php-version: ['8.1', '8.2', '8.3', '8.4']
```

Comment:
Mh, latest workflow from nextcloud/.github should get this automatically from the supported nextcloud versions, I believe

Reply:
It's now based on the nextcloud/.github psalm-matrix one. I just changed the "on" condition to only run on PRs for some branches and if some files where changed.